src/main/resources/GoogleApps/JSExtension.xml

```diff
@@ -111,50 +122,212 @@
         <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
       </use>
     </class>
+    <property>
+      <cache>short</cache>
+    </property>
+    <property>
+      <code>define('googleapps-buttons', {
+  prefix: 'googleapps.',
+  keys: [
+    'edit.editingoogleapps.link',
+    'import.importfromgoogleapps'
+  ]
+});
+require(['jquery', 'xwiki-l10n!googleapps-buttons'], function ($, l10n) {
+  var listener = function () {
+    var hasPagination = $("#_attachments table").length &gt; 0;
+    $("#_attachments span.name").each(function (i, elem) {
+      var searchPoint = hasPagination ? $(elem).closest('tr').find('.attachmentActions') : $(elem.parentNode).find('.xwikibuttonlinks');
+      if ($(searchPoint).find('a.editlink').length) {
+        return;
+      }
+      var dlink = hasPagination ? $(searchPoint).find(".actiondelete") : $(searchPoint).find(".deletelink");
+      if (dlink.length &gt; 0) {
+        var dlinkurl = dlink[0].href;
+        var dlinkpos = dlinkurl.indexOf("?");
+        var filename = dlinkurl.substring(dlinkurl.lastIndexOf("/", dlinkpos) + 1, dlinkpos);
+        var googleDoc = new XWiki.Document('EditInGoogleApps', 'GoogleApps');
+        var queryParams = {
+          page: encodeURIComponent(XWiki.currentSpace) + '.' + encodeURIComponent(XWiki.currentPage),
+          name: filename
+        };
+        searchPoint.append('&lt;a class="editlink" href="' + googleDoc.getURL('view', $.param(queryParams)) + '" title="' + l10n.get('edit.editingoogleapps.link') + '"&gt;' + l10n.get('edit.editingoogleapps.link') + '&lt;/a&gt;');
+      }
+    });
+    var attachaddform = $("#AddAttachment");
+    if (attachaddform.find('.googleimport').length) {
+      return;
+    }
+    var googleDoc = new XWiki.Document('ImportFromGoogleApps', 'GoogleApps');
+    attachaddform.append('&lt;span class="buttonwrapper googleimport" style="float: right; position: relative; top: -25px;"&gt;&lt;a href="' + googleDoc.getURL('view', $.param({ page: encodeURIComponent(XWiki.currentSpace) + '.' + encodeURIComponent(XWiki.currentPage) })) + '"&gt;' + l10n.get('import.importfromgoogleapps') + '&lt;/a&gt;&lt;/span&gt;');
```

Comment:
Same as above, it's safer to set attribute / text using the API instead of string concatenation. Also, use the API to get the reference of the current page.